src/Conversion/ONNXToTorch/NN/Reshape.cpp

```diff
@@ -0,0 +1,90 @@
+
+//===------- ReshapeOp.cpp - ONNX Op Transform ------------------===//
+// ===================================================================
+//
+// This file implements a combined pass that dynamically invoke several
+// transformation on ONNX ops.
+//
+//===-------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToTorch/ONNXToTorchCommon.hpp"
+#include "torch-mlir/Dialect/Torch/IR/TorchOps.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchTypes.h"
+#include "llvm/ADT/ArrayRef.h"
+
+using namespace mlir;
+using namespace mlir::torch;
+
+static llvm::Optional<std::pair<unsigned, unsigned>> isEquivalentToFlatten(const llvm::ArrayRef<int64_t> &inputShape, const llvm::ArrayRef<int64_t> &targetShape) {
+  llvm::Optional<std::pair<unsigned, unsigned>> savedResult;
+  int dimDiff = inputShape.size() - targetShape.size(); 
```

Comment:
nit: `unsigned` or `size_t` for consistency.

```suggestion
  unsigned dimDiff = inputShape.size() - targetShape.size(); 
```

Reply:
Hmmm, but that can create an overflow if for some reasons `targetShape` is bigger than `inputShape` isn't is ?

Reply:
Sorry, you are right. Reviewed that a bit too quickly on Friday.